views.py

```diff
@@ -66,13 +67,19 @@ def convert(request, article_id=None, file_id=None):
         stripped_path, exten = os.path.splitext(orig_path)
 
         if exten not in ['.docx', '.rtf']:
-            raise TypeError('Unexpected Manuscript File Type')
+            messages.add_message(request, messages.ERROR, 'The Pandoc plugin currently only supports .docx and .rtf filetypes')
+            return HttpResponseRedirect(request.META.get('HTTP_REFERER'))
 
         temp_md_path = stripped_path + '.md'
 
         # construct and execute subprocess.run() command to create intermediate md file
         pandoc_command = base_pandoc_command + ['-s', orig_path, '-t', 'markdown', '-o', temp_md_path]
-        subprocess.run(pandoc_command, stderr=subprocess.PIPE, check=True)
+        try:
+            subprocess.run(pandoc_command, stderr=subprocess.PIPE, check=True)
+        except subprocess.CalledProcessError as e:
+            messages.add_message(request, messages.ERROR, 'Pandoc encountered the following error when executing the command {cmd}: {err}'.format(cmd=e.cmd, err=e.stderr))
```

Comment:
One ? I had as a bit of a python newbie - it looks like `subprocess.run` captures the output error as a byte string, so right now the message is wrapped with `b"..."`. I tried to add `decode("utf-8")` onto `e.stderr` but this seemed to stop the toast from showing up at all... any ideas what's going on there?

Reply:
Yeah this is caused by a newline in the stderr, I've added an Issue and PR to fix:

https://github.com/BirkbeckCTP/janeway/issues/920
and
https://github.com/BirkbeckCTP/janeway/pull/921